beacon_node/store/src/leveldb_store.rs

```diff
@@ -147,10 +148,30 @@ impl<E: EthSpec> Store<E> for LevelDB<E> {
         SimpleForwardsBlockRootsIterator::new(store, start_slot, end_state, end_block_root)
     }
 
-    fn do_atomically(&self, ops_batch: &[StoreOp]) -> Result<(), Error> {
+    fn do_atomically(&self, ops_batch: &[StoreOp<E>]) -> Result<(), Error> {
         let mut leveldb_batch = Writebatch::new();
         for op in ops_batch {
             match op {
+                StoreOp::PutBlock(block_hash, block) => {
+                    let untyped_hash: Hash256 = (*block_hash).into();
+                    let key = Self::get_key_for_col(
+                        DBColumn::BeaconBlock.into(),
+                        untyped_hash.as_bytes(),
+                    );
+                    let value = block.as_store_bytes();
+                    leveldb_batch.put(key, &value);
+                }
+
+                StoreOp::PutState(state_hash, state) => {
+                    let untyped_hash: Hash256 = (*state_hash).into();
+                    let key = Self::get_key_for_col(
+                        DBColumn::BeaconState.into(),
+                        untyped_hash.as_bytes(),
+                    );
+                    let value = StorageContainer::new(state).as_ssz_bytes();
+                    leveldb_batch.put(key, &value);
+                }
```

Comment:
State storage needs to take into account the `HotStateSummary` here, as in the non-atomic case:

https://github.com/sigp/lighthouse/blob/d79e07902e8d3daf38fb79cd43436d513ca22675/beacon_node/store/src/hot_cold_store.rs#L372-L398

The compositionality of the `HotColdDB` consisting of two sub-stores (`hot_db` and `cold_db`) with the same API _is a lie_, and is another reason why the `Store` trait should be refactored IMO. I don't think `Store` should be implemented for `LevelDB`, because we never use just one `LevelDB` database as the store any more, and it creates awkwardness when a `Store` API has to be implemented on the underlying store with assumptions about that API only getting used for the hot or cold DB (in this case we're assuming hot). I think it makes sense to put all this logic in the parent impl of `do_atomically` (i.e. `HotColdDB`), and stub out the impl for `LevelDB` so that it returns an error.